pkg/runtime/reconciler.go

```diff
@@ -292,12 +288,16 @@ func (r *resourceReconciler) ensureConditions(
 	exit := rlog.Trace("r.ensureConditions")
 	defer exit(err)
 
-	if syncedCond := ackcondition.Synced(res); syncedCond == nil {
-		rlog.Debug("resource missing ACK.ResourceSynced condition")
-		// only the resource manager will know whether the resource is in a
-		// stable sync state. Even if we got no error back from the
-		// create/update operations, we can only set this to Unknown.
-		condStatus := corev1.ConditionUnknown
+	// If the ACK.ResourceSynced condition is not set using the custom hooks,
+	// determine the Synced condition using "rm.IsSynced" method
+	if ackcondition.Synced(res) == nil {
+		condStatus := corev1.ConditionFalse
+		synced := false
+		rlog.Enter("rm.IsSynced")
+		if synced, err = rm.IsSynced(ctx, res); err == nil && synced {
```

Comment:
Should we also store the err in a condition if `IsSynced` fails?

Reply:
Good idea, it is helpful from user perspective to see the error message in Condition message.